src/pages/Cart/Cart.test.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import renderWithRedux from '../../testUtils/renderWithRedux.js';
+import Cart from './Cart';
+
+describe('Cart', () => {
+  it('starts loading cart', () => {
+    const { getByTestId } = renderWithRedux(<Cart />, {
+      initialState: {
+        cart: {
+          items: [],
+          isEmptyCartSet: false,
+          numberOfItems: 0,
+          totalSum: 0,
+          loading: true,
+        },
+      },
+    });
+    const loader = getByTestId('loader');
+    const numberOfItemsInCart = getByTestId('number-of-cart-items');
+    const cartSum = getByTestId('cart-sum');
+    const finishOrderBtn = getByTestId('finish-order-btn');
+
+    expect(loader).toBeInTheDocument();
+    expect(numberOfItemsInCart).toHaveTextContent('0');
+    expect(cartSum).toHaveTextContent('0.00 $');
+    expect(finishOrderBtn).toBeDisabled();
+  });
+
+  it('renders empty cart', () => {
+    const { getByTestId, getByText, queryByTestId } = renderWithRedux(
+      <Cart />,
+      {
+        initialState: {
+          cart: {
+            items: [],
+            isEmptyCartSet: true,
+            numberOfItems: 0,
+            totalSum: 0,
+            loading: false,
+          },
+        },
+      }
+    );
+    const loader = queryByTestId('loader');
+    const heading = getByText(/Your cart is empty/i);
+    const numberOfItemsInCart = getByTestId('number-of-cart-items');
+    const cartSum = getByTestId('cart-sum');
+    const finishOrderBtn = getByTestId('finish-order-btn');
+
+    expect(loader).toBeNull();
+    expect(heading).toBeInTheDocument();
+    expect(numberOfItemsInCart).toHaveTextContent('0');
+    expect(cartSum).toHaveTextContent('0.00 $');
+    expect(finishOrderBtn).toBeDisabled();
+  });
+
+  it('renders cart with items', () => {
+    const { getByTestId, getAllByTestId } = renderWithRedux(<Cart />, {
+      initialState: {
```

Comment:
Is this state correct? I see the items but `isEmptyCartSet: true`.

Reply:
you right, solved and reworked here https://github.com/maxSedelnikov/react-shopping-app/pull/6/commits/c759e7206dd9006634852446e70ed64047e50a1d